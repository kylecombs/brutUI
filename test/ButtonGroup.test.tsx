import * as React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { ButtonGroup, options1 } from "../src";
import { FiGrid, FiList } from "react-icons/fi";

afterEach(cleanup);

describe("ButtonGroup", () => {
  it("Renders correctly (no icons)", () => {
    const { asFragment, getByText } = render(
      <ButtonGroup active="Leading" setActive={() => {}} options={options1} />,
    );

    expect(getByText("Leading")).toHaveClass("bg-gray-50 dark:bg-gray-700");
    expect(getByText("Middle")).not.toHaveClass("bg-gray-50 dark:bg-gray-700");
    expect(getByText("Trailing")).not.toHaveClass(
      "bg-gray-50 dark:bg-gray-700",
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders correctly (icons)", () => {
    const { asFragment } = render(
      <ButtonGroup
        active="list"
        setActive={() => {}}
        options={[
          {
            content: <FiList size={20} />,
            value: "list",
          },
          {
            content: <FiGrid size={20} />,
            value: "grid",
          },
        ]}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Clicking on different option applies active classes", () => {
    let active = "Leading";

    const { rerender, getByText } = render(
      <ButtonGroup
        active={active}
        setActive={(newActive: string) => {
          active = newActive;
        }}
        options={options1}
      />,
    );

    fireEvent.click(getByText("Middle"));

    rerender(
      <ButtonGroup active={active} setActive={() => {}} options={options1} />,
    );

    expect(getByText("Leading")).not.toHaveClass("bg-gray-50 dark:bg-gray-700");
    expect(getByText("Middle")).toHaveClass("bg-gray-50 dark:bg-gray-700");
    expect(getByText("Trailing")).not.toHaveClass(
      "bg-gray-50 dark:bg-gray-700",
    );
  });
});
