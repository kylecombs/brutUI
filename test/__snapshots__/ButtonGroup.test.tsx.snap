// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Renders correctly (icons) 1`] = `
<DocumentFragment>
  <button
    class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-l-lg border-r-0 bg-gray-50 dark:bg-gray-700"
  >
    <svg
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="8"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="8"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="8"
        x2="21"
        y1="18"
        y2="18"
      />
      <line
        x1="3"
        x2="3.01"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="3.01"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="3.01"
        y1="18"
        y2="18"
      />
    </svg>
  </button>
  <button
    class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-r-lg"
  >
    <svg
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="7"
        width="7"
        x="3"
        y="3"
      />
      <rect
        height="7"
        width="7"
        x="14"
        y="3"
      />
      <rect
        height="7"
        width="7"
        x="14"
        y="14"
      />
      <rect
        height="7"
        width="7"
        x="3"
        y="14"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`ButtonGroup Renders correctly (no icons) 1`] = `
<DocumentFragment>
  <button
    class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-l-lg border-r-0 bg-gray-50 dark:bg-gray-700"
  >
    Leading
  </button>
  <button
    class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 border-r-0"
  >
    Middle
  </button>
  <button
    class="inline-flex whitespace-nowrap items-center h-10 px-4 font-medium text-sm focus:outline-none hover:bg-gray-50 dark:hover:bg-gray-700 text-gray-700 dark:text-white border border-gray-300 dark:border-gray-500 rounded-r-lg"
  >
    Trailing
  </button>
</DocumentFragment>
`;
